data_controller.rb

```diff
@@ -0,0 +1,90 @@
+require './book'
+require './rental'
+require './student'
+require './teacher'
+require 'json'
+
+module DataController
+  def get_book_by_title(title)
+    @books.find { |book| book.title == title }
+  end
+
+  def get_person_by_id(id)
+    @persons.find { |person| person.id == id.to_i }
+  end
+
+  def load_books
+    load_data('books.json') { |element| Book.new(element['title'], element['author']) }
+  end
+
+  def load_people
+    load_data('people.json') do |element|
+      if element['data_type'] == 'Teacher'
+        Teacher.new(
+          id: element['id'].to_i,
+          age: element['age'].to_i,
+          specialization: element['specialization'],
+          name: element['name']
+        )
+      else
+        Student.new(
+          id: element['id'].to_i,
+          age: element['age'].to_i,
+          name: element['name'],
+          parent_permission: element['parent_permission']
+        )
+      end
+    end
+  end
+
+  def load_rentals
+    load_data('rentals.json') do |element|
+      person = get_person_by_id(element['person_id'])
+      book = get_book_by_title(element['book_title'])
+      Rental.new(element['date'], book, person)
+    end
+  end
+
+  def save_books
+    data = @books.map { |book| { title: book.title, author: book.author } }
+    save_data('books.json', data)
+  end
+
+  def save_people
+    data = @persons.map do |person|
+      {
+        id: person.id,
+        age: person.age,
+        name: person.name,
+        data_type: person.class,
+        **person.to_s
+      }
+    end
+    save_data('people.json', data)
+  end
+
+  def save_rentals
+    data = @rentals.map do |rental|
+      {
+        date: rental.date,
+        book_title: rental.book.title,
+        person_id: rental.person.id
+      }
+    end
+    save_data('rentals.json', data)
+  end
+
+  private
+
+  def load_data(file, &block)
+    if File.exist?(file) && !File.empty?(file)
+      JSON.parse(File.read(file)).map(&block)
+    else
+      []
+    end
+  end
+
+  def save_data(file, data)
+    File.write(file, JSON.generate(data))
+  end
```

Comment:
- [ ] The app works well, I was able to create a person, book, and rental, but looking at the requirements for this project, data wasn't saved. After exiting the app using option 7, data should be saved and then loaded when the app is started. I have attached an image for your reference. You may as well fix the error that comes when a user use option 7 (see attached image).

![save](https://user-images.githubusercontent.com/53124548/225607502-14f79a31-e609-49d1-a9f4-1baa4b557388.png)